### PR TITLE
chore(content-releases): show Banner reached limit max releases

### PR DESCRIPTION
**Pull request by simotae14**
### What does it do?

It shows a banner in the Content releases page when you reach the maximum number of pending releases of your plan.
In addition, we disable the "New Release" button.

### Why is it needed?

Because the cloud plans have a limit on the number of pending content releases

### Related issue(s)/PR(s)

CS-453

### CODE REVIEW

File: packages/core/content-releases/admin/src/pages/ReleasesPage.tsx
Comment (Feranchz) [inline packages/core/content-releases/admin/src/pages/ReleasesPage.tsx:218]: I think you can change this logic to:

```
const { maximumReleases = 3 } = getFeature('cms-content-releases');
```

and then you can do 

```
const hasReachedMaximumPendingReleases = totalReleases >= maximumReleases
```

File: packages/core/admin/shared/contracts/admin.ts
Comment (markkaylor) [inline packages/core/admin/shared/contracts/admin.ts:176]: I think this is incomplete and should be something closer to. Technically a number should always be defined but I think it is possible that someone could make a mistake when configuring the license in which case it would be null.

```suggestion
  name: 'cms-content-releases';
  options?: {
  		maximumReleases: number | null;
  }
```

File: packages/core/content-releases/admin/src/pages/ReleasesPage.tsx
Comment (markkaylor) [inline packages/core/content-releases/admin/src/pages/ReleasesPage.tsx:320]: nitpick/observation not a request for change per se: I notice sometimes in the codebase styled components use either a prefix Custom or a prefix Styled. I think a convention that is trending right now in the codebase is to refer to styled components with the Styled prefix instead of the Custom prefix. I think it would be nice if we choose one and we are consistent. You don't have to change this though if you don't want.

File: packages/core/content-releases/admin/src/translations/en.json
Comment (markkaylor) [inline packages/core/content-releases/admin/src/translations/en.json:20]:
```suggestion
  "pages.Releases.max-limit-reached.title": "You have reached the {number} pending {number, plural, one {release} other {releases}} limit.",
```

File: packages/core/content-releases/admin/src/pages/ReleasesPage.tsx
Comment (markkaylor) [inline packages/core/content-releases/admin/src/pages/ReleasesPage.tsx:335]:
```suggestion
                    'You have reached the {number} pending {number, plural, one {release} other {releases}} limit.',
```

File: packages/core/content-releases/admin/src/pages/ReleasesPage.tsx
Comment (markkaylor) [inline packages/core/content-releases/admin/src/pages/ReleasesPage.tsx:321]: I don't think the designs have this marginTop, only marginBottom

Comment (simotae14) [reply packages/core/content-releases/admin/src/pages/ReleasesPage.tsx:321]: To be honest I think we need the marginTop by design
here you can see the solution without marginTop
withoutMarginTop
and here you can see the spaces by design
figma

Comment (markkaylor) [reply packages/core/content-releases/admin/src/pages/ReleasesPage.tsx]: I think there is another issue at play here. The way I interpret the designs there is 40px marginBottom in the Header. The distance from 3 releases to the banner displaying the release limit should be that marginBottom, 40px. So there should be no need for marginTop except for some reason the banner is partially inside the Header. I think we should investigate why that is. Applying marginTop feels like covering up an issue. 

Screenshot 2024-01-26 at 17 57 12

Comment (simotae14) [reply packages/core/content-releases/admin/src/pages/ReleasesPage.tsx]: Found the bug, inside the HeaderLayout component we have a height for the wrapper calculated based on the content, https://github.com/strapi/design-system/blob/ab8326385e52663ec898814b30d8beeda37b1659/packages/strapi-design-system/src/Layout/HeaderLayout.tsx#L46, in our case the height was calculated with all the content without subtitle because it is filled with content coming from an API. To avoid this I have now added a default subtitle content "No releases" when we are waiting for the API response

File: packages/core/content-releases/admin/src/translations/en.json
Comment (markkaylor) [inline packages/core/content-releases/admin/src/translations/en.json:20]: You don't need this anymore right?

**Review by markkaylor — APPROVED** [review]
Great looks good! I just think you have a lingering translation after the refactor. Please remove it before you merge.

File: packages/core/content-releases/admin/src/translations/en.json
Comment (simotae14) [reply packages/core/content-releases/admin/src/translations/en.json:20]: yeah my bad

**Pull request merged**
